### ISSUE TITLE
Webhook production url does not work on initial load

### ISSUE THREAD

**Issue by sdevalapurkar** [n8n-io/n8n#2155]
**Describe the bug**
When using a dockerized setup of n8n with an nginx proxy, and with local credential and workflow files being imported on load, even though the workflow is active the production url does not work and gives a CORS error:

![image](https://user-images.githubusercontent.com/28017034/131540958-6426fc94-ce28-43bc-ade5-8828bb0a73e4.png)

![image](https://user-images.githubusercontent.com/28017034/131541047-d802ccee-a299-4f29-871b-e196ababdbe5.png)

Our nginx config for n8n:

```
server {
  listen 5100;

  location / {
    proxy_pass http://n8n:5678;
    proxy_redirect default;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $server_name;

    if ($request_method = 'OPTIONS') {
      add_header "Access-Control-Allow-Origin" "*";
      add_header "Access-Control-Allow-Methods" "GET, POST, OPTIONS, HEAD";
      add_header "Access-Control-Allow-Headers" "Authorization, Origin, X-Requested-With, Content-Type, Accept, sessionid";
      return 204;
    }
  }
}
```

We also set response headers in the webhook to try and avoid cors errors:

![image](https://user-images.githubusercontent.com/28017034/131541615-eafcb9df-5515-4620-b7e9-cc2f06df7a20.png)

Interestingly, if we go to n8n (localhost:5678) and toggle the active state of the workflow to off and then to on, then the production url for the workflow starts working.

**Expected behavior**
Production url for workflow should work on initial load without CORS errors and without the need to toggle active state to off then on.

**Environment (please complete the following information):**
 - Running in docker
 - n8n Version ^0.135.2
 - Node.js Version >= 14
 - Database system Postgres

**Comment by janober** [comment]
Yes, sorry that is currently sadly expected behavior and also strictly speaking not a bug.
The reason is that the CLI tool was written for backup & database migration, it was never meant to be used when another n8n process runs at the same time.

The technical reason why it currently does not work: There is a special database table in which active webhooks get saved. It writes entries in there on startups (for all the active workflows) and every time a workflow gets activated. This entry does currently not get created and for that reason can it not find it.

So the easiest thing to do would be to restart n8n after you did import the data.

Hope that is helpful.

**Comment by sdevalapurkar** [comment]
> Yes, sorry that is currently sadly expected behavior and also strictly speaking not a bug.
> The reason is that the CLI tool was written for backup & database migration, it was never meant to be used when another n8n process runs at the same time.
> 
> The technical reason why it currently does not work: There is a special database table in which active webhooks get saved. It writes entries in there on startups (for all the active workflows) and every time a workflow gets activated. This entry does currently not get created and for that reason can it not find it.
> 
> So the easiest thing to do would be to restart n8n after you did import the data.
> 
> Hope that is helpful.

I see, ok thank you for the quick response. I guess we will have to use the workaround of restarting n8n after the import for now. If there is any update on this in future, please let us know. Cheers
